flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/logical/utils/LogicalTypeChecks.java

```diff
@@ -332,4 +351,162 @@ public Boolean visit(DayTimeIntervalType dayTimeIntervalType) {
 			}
 		}
 	}
+
+	private static class TypeEquivalenceIgnoreNamesExtractor extends Extractor<Boolean> {
```

Comment:
Should directly extend from `LogicalTypeDefaultVisitor` because we override the default method anyway.